src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -793,6 +793,36 @@ R"(#include <stdio.h>
         }
     }
 
+    void visit_StringSection(const ASR::StringSection_t& x) {
+        self().visit_expr(*x.m_arg);
+        std::string arg, left, right, step, left_present, rig_present;
+        arg = src;
+        if (x.m_start) {
+            self().visit_expr(*x.m_start);
+            left = src;
+            left_present = "true";
+        } else {
+            left = "0";
+            left_present = "false";
+        }
+        if (x.m_end) {
+            self().visit_expr(*x.m_end);
+            right = src;
+            rig_present = "true";
+        } else {
+            right = "0";
+            rig_present = "false";
+        }
+        if (x.m_step) {
+            self().visit_expr(*x.m_step);
+            step = src;
+        } else {
+            step = "1";
+        }
+        src = "_lfortran_str_slice(" + arg + ", " + left + ", " + right + ", " + \
```

Comment:
Do we need to handle the case where both `x.m_start`, `x.m_end` and `x.m_step` are `NULL`? Do we test this? I think there will be 8 cases to test with `nullptr` and `not nullptr` for each `x.m_start`, `x.m_end` and `x.m_end` giving 2 x 2 x 2 = 8 cases. Are we making sure we test each of these. If some of these are supposed to give semantic error then include them in `errors/` folder.

Reply:
Done